py-rse/style.Rmd

```diff
@@ -338,65 +386,115 @@ if __name__ == '__main__':
     print('number', n)
 ```
 
-## How should I collaborate with people when doing code reviews? {#py-rse-style-how}
+## How should I review someone else's code? {#py-rse-style-review}
 
-FIXME: how to review on GitHub.
+The biggest benefit of having a second person work on a programming project
+is therefore not getting twice as much code written,
+but having code reviewed.
+Study after study over more than 40 years has shown that code review is the most effective way to find bugs in software
+@Faga1976,@Faga1986,@Cohe2010,@Bacc2013.
+Despite this,
+it still isn't common in research software development,
+in part because it isn't part of the culture @Sega2005,
+but also because code review is mostly useful
+when the reviewers understand the problem domain well enough to comment on algorithms and design choices
+rather than indentation and variable naming,
+and the number of people who can do that for a research project is typically very small—sometimes
+as small as one @Petr2014.
 
-## In what order should functions be defined? {#py-rse-style-order}
+This lesson will look at the mechanics of code review
+and present some short examples of the kinds of things reviewers should look for.
 
-When encountering code for the first time,
-most people scan it from top to bottom.
-If that code is a program or script,
-rather than a library,
-its main function should be put first,
-and should probably be called `main`.
+How you review is just as important as what you look for:
+being dismissive or combative are good ways to ensure that people don't pay attention to your reviews,
+or avoid having you review their work.
+Equally,
+being defensive when someone offers suggestions politely and sincerely is very human,
+but can stunt your development as a programmer.
 
-After reading that function,
-someone should have a good idea of what the program does in what order.
-Three common patterns that people might match against are:
+Lots of people have written guidelines for doing reviews,
+which are also useful when reviewing written work @Quen2018,@Sank2018.
+A few key points are:
 
-1.  Figure out what the user has asked it to do (Chapter \@ref(configure)).
-2.  Read all input data.
-3.  Process it.
-4.  Write output.
+Work in small increments.
+:   As @Cohe2010 and others have found,
+    code review is most effective when done in short bursts.
+    That means that change requests should also be short:
+    anything that's more than a couple of screens long
+    should be broken into smaller pieces.
 
-or:
+Look for algorithmic problems first.
```

Comment:
Somewhere it would be good to add a note that following coding coding style helps you reviewers focus on algorithmic problems.

Reply:
done.